test/watershed.jl

```diff
@@ -0,0 +1,22 @@
+using ImageFiltering
+
+@testset "watershed" begin
+    
+    img = zeros(100, 100)
+    img = zeros(Float64, (100,100))
+    img[25:75, 25:75] = 1
+
+    img, _ = magnitude_phase(img)
```

Comment:
Watershed segmentation is usually used with distance transform(e.g, demo below) or gradient image of the original image (e.g, [here](https://www.researchgate.net/profile/Serge_Beucher/publication/233950923_Segmentation_The_Watershed_Transformation_Mathematical_Morphology_in_Image_Processing/links/55f7c6ce08aeba1d9efe4072/Segmentation-The-Watershed-Transformation-Mathematical-Morphology-in-Image-Processing.pdf) page 13). In the topographic surface of the gradient image, the region boundary becomes the watershed whereas the topographic surface of original image is just two flat regions.

Did you want me to directly initialize ```img``` as the gradient image?